drivers/usb/udc/udc_stm32.c

```diff
@@ -985,28 +989,55 @@ static const struct udc_stm32_config udc0_cfg  = {
 	.speed_idx = DT_ENUM_IDX_OR(DT_DRV_INST(0), maximum_speed, 1),
 };
 
+/*
+ * Function to determine the USB PHY interface selection for STM32 using device tree configurations.
+ * The function checks for compatible PHY types and phy phandle,
+ * and returns the corresponding physel.
+ * There are three types of supported physel:
+ * - USB_OTG_HS_PHY_ULPI: UTMI+ Low Pin Interface for external high-speed PHY.
+ * - USB_FS_EMBEDDED_PHY: on-chip full-speed PHY.
+ * - USB_OTG_HS_EMBEDDED_PHY: on-chip high-speed PHY.
+ */
+static uint8_t usb_dc_stm32_get_physel(void)
+{
+	uint8_t physel;
+
+#if DT_HAS_COMPAT_STATUS_OKAY(usb_ulpi_phy)
+	physel = USB_OTG_HS_PHY_ULPI;
```

Comment:
I'd add a comment here stating that if "usb-ulpi-phy" is enabled on board.dts, we suppose it is selected (otherwise it doesn't make no sense to define it).